examples/cms/src/App.tsx

```diff
@@ -0,0 +1,134 @@
+    }
+
+    useLayoutEffect(() => {
+        if (!selectedDataSource) {
+            framer.showUI({
+                width: UI_DEFAULTS.SETUP_WIDTH,
+                height: UI_DEFAULTS.SETUP_HEIGHT,
+                resizable: false,
+            })
+            return
+        }
+
+        framer.showUI({
+            width: UI_DEFAULTS.MAPPING_WIDTH,
+            height: UI_DEFAULTS.MAPPING_HEIGHT,
+            minWidth: UI_DEFAULTS.MAPPING_WIDTH,
+            minHeight: UI_DEFAULTS.MAPPING_HEIGHT,
+            resizable: true,
+        })
+    }, [selectedDataSource])
+
+    if (result.status === "needsSetup" && !selectedDataSource) {
+        return (
+            <main className="setup">
+                <div className="logo">
+                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 150 150">
+                        <path
+                            fill="#999"
+                            d="M75 33c18.778 0 34 7.611 34 17S93.778 67 75 67s-34-7.611-34-17 15.222-17 34-17Zm34 40.333C109 82.538 93.778 90 75 90s-34-7.462-34-16.667V60c0 9.389 15.222 17 34 17 18.776 0 33.997-7.61 34-16.997v13.33ZM109 84v.497c0-.166-.005-.332-.015-.497Zm0 13.333C109 106.538 93.778 114 75 114s-34-7.462-34-16.667V84h.015c-.01.166-.015.333-.015.5 0 9.113 15.222 16.5 34 16.5 18.776 0 33.997-7.386 34-16.497v12.83Z"
+                        />
+                    </svg>
+                </div>
+                <form onSubmit={showFieldsMapping}>
+                    <label htmlFor="collection">
+                        Collection
+                        <select
+                            id="collection"
+                            onChange={event => setSelectedDataSourceId(event.target.value)}
+                            value={selectedDataSourceId || ""}
+                        >
+                            <option value="" disabled>
+                                Choose...
+                            </option>
+                            {result.allDataSources.map(collectionId => (
+                                <option key={collectionId} value={collectionId}>
+                                    {collectionId}
+                                </option>
+                            ))}
+                        </select>
+                    </label>
+                    <button style={{ position: "relative" }} disabled={!selectedDataSourceId || isLoadingFields}>
+                        {isLoadingFields ? <Spinner inheritColor /> : "Next"}
+                    </button>
+                </form>
+            </main>
+        )
+    }
+
+    if (selectedDataSource) {
+        return (
+            <FieldMapping
+                dataSource={selectedDataSource}
+                savedFieldsConfig={result.status === "needsConfiguration" ? result.savedFieldsConfig : null}
+                existingFields={existingFields}
+                savedSlugFieldId={syncSlugFieldId}
+                onSubmit={syncCollection}
+            />
+        )
```

Comment:
This is nicely abstracted in a component, shall we do the same for data source selection? Then the parent component just decides one or the other. Would make it easier to see a clear overview of the different UI we render